### ISSUE TITLE
Error deserializing SwiftLint JSON response" when using Xcode 14.0+ on Danger report

### ISSUE THREAD

**Issue by sachinbhardwaj-tul-ios** [realm/SwiftLint#5022]
<img width="824" alt="Screenshot 2023-05-19 at 6 07 07 PM" src="https://github.com/realm/SwiftLint/assets/90173600/d40590a0-a7c9-4380-baf9-23b23889ae43">
Please let us know how to fix that

**Comment by SimplyDanny** [comment]
I'm rather unfamiliar with the Danger use cases, unfortunately. Would you tell me how you run SwiftLint (including the used configuration) and on which file this error is shown?

**Comment by SimplyDanny** [comment]
Is this still an issue for you. @sachinbhardwaj-tul-ios?

**Comment by sachinbhardwaj-tul-ios** [comment]
yes

**Comment by dsilva-adastra** [comment]
Hello everyone!

I have this issue. I am not sure if it's an issue in Danger or Swiftlint. What I noticed is that Danger doesn't know about the Swiftlint excluded files and it is still processing them and since Swiftlint returns nothing we get that error. I am only guessing here.

I came up with a logic to avoid those errors if anyone would be interested

```swift
func checkSwiftLintIssues() {

    // File names or folders that should be ignored. 
    // The strings I added below are only examples
    let pathsFilesToIgnore =
    [
        "File.swift",
        "Path/To/File"
    ]
    
    // Filter out anything that is not a swift file and whether the name of a file contains any of those items inside the pathsFilesToIgnore
    let modifiedCreatedFilesFiltered = (danger.git.modifiedFiles + danger.git.createdFiles)
        .filter { $0.fileType == .swift }
        .filter { file in pathsFilesToIgnore.first { file.name.contains($0) } == nil }

    guard modifiedCreatedFilesFiltered.isEmpty == false else {
        return
    }

    let lintIssues = SwiftLint
        .lint(.modifiedAndCreatedFiles(directory: directory))

    if lintIssues.isEmpty == false {
        let issues = lintIssues.count == 1 ? "1  issue" : "\(lintIssues.count) issues"

        warn(
            "‼️ SwiftLint found \(issues) (excluding TODOs)"
        )
    } else {
        message(
            "We are in the clear. No SwiftLint issues found!"
        )
    }
}
```
